gui/web/src/App.js

```diff
@@ -40,6 +46,7 @@ class App extends Component {
 
   componentDidMount() {
     this.setVersion()
+    console.log(this.objects.metricsTracker.sendEvent("ce:test_startup"));
```

Comment:
we can just send the event, no need to log to console unless for local dev debugging.

Reply:
we should add the necessary startup items here, like we do in the CLI. don't we need to code that up in metricsTracker.js too?

Reply:
Yes, we do - but shouldn't those be added to the constructor, as we do in the CLI, rather than per-request?

Reply:
I was thinking about this a little and it's actually better to have a very different pattern. Let's reuse the infrastructure we have on the backend for this. We should create a new endpoint on the backend called `/sendMetricEvent` which takes in a JSON object as data, which it can add to the event props. It will reuse all the common props (other than 1 bit to distinguish this is come from the UI admin page)

the frontend would not need any metric specific logic and would just have one sendEvent.js function call that it can use to send events.

This will streamline all our metrics through one backend endpoint and common infrastructure which will make it much much less painful if we ever decide to switch providers or even do something like adding decorators on event tracking. This also moves a lot of the logic to the backend which is preferable